workspaces/sandbox/packages/app/e2e-tests/app.test.ts

```diff
@@ -14,14 +14,118 @@
  * limitations under the License.
  */
 
-import { test, expect } from '@playwright/test';
+import { test, expect, Page, type Locator } from '@playwright/test';
+import { LoginPage } from '../Utils/loginPage';
 
-test('App should render the welcome page', async ({ page }) => {
-  await page.goto('/');
+const dotenv = require('dotenv');
 
-  const enterButton = page.getByRole('button', { name: 'Enter' });
-  await expect(enterButton).toBeVisible();
-  await enterButton.click();
+dotenv.config(); // Load environment variables from .env file
 
-  await expect(page.getByText('My Company Catalog')).toBeVisible();
+let page: Page;
+
+// Load credentials and base URL from environment variables
+const id = process.env.PLAYWRIGHT_ID!;
+const pw = process.env.PLAYWRIGHT_PW!;
+const baseUrl = process.env.BASE_URL!;
+
+test.describe('sandbox plugin', () => {
+  // Before all tests: launch browser, create new page, and log in
+  test.beforeAll('', async ({ browser }) => {
+    const context = await browser.newContext({
+      viewport: { width: 1700, height: 940 }, // e.g. MacBook Pro screen resolution (CSS px)
+    });
+    page = await context.newPage();
+
+    const login_page = new LoginPage(page);
+    await login_page.navigate(baseUrl); // Navigate to base URL
+    await login_page.login(id, pw); // Log in with credentials
+  });
+
+  // Test the homepage layout and welcome text
+  test('Test Homepage', async () => {
+    await page.locator('a').filter({ hasText: 'Home' }).isVisible();
+
+    // Verify ARIA snapshot for the main article content
+    await expect(page.getByRole('article')).toMatchAriaSnapshot(`
+      - img "Red Hat Trial"
+      - heading "Try Red Hat products" [level=1]
+      - paragraph: Explore, experiment, and see what's possible
+      - paragraph: /Click on "Try it" to initiate your free, no commitment \\d+-day trial\\./
+    `);
+  });
+
+  // Test clicking the OpenShift card on the homepage
+  test('Test Homepage Openshiftcard', async () => {
+    const card = page.getByText('OpenShift Comprehensive cloud');
+    await expect(card).toBeVisible(); // Ensure card is visible
+    await card.getByRole('button', { name: 'Try it' }).click();
+    // Verify expected text on the resulting article page
+    await expect(page.getByRole('article')).toContainText('Welcome');
+    await expect(page.getByRole('article')).toContainText(
+      'Your free trial expires in 30 days',
```

Comment:
```suggestion
      'Click on "Try it" to initiate your free, no commitment 30-day trial.',
```

It is failing for me here. When I log in for the first time, this is what appears to me:
<img width="1681" alt="image" src="https://github.com/user-attachments/assets/1c2cc7e4-ec23-4633-b714-823924f96e6a" />
When I click for the first time on "Try it", this is what I get:
<img width="1650" alt="image" src="https://github.com/user-attachments/assets/07ae6bc9-4d45-4af5-b290-25c16b59e565" />

CC: @mfrancisc

Reply:
The flow is different in the e2e-test environment. Since we automatically approve, this is what is happening: https://drive.google.com/file/d/1fHfEXhnJg9sClsG7Aplz-ndX6-2TWOdr/view?usp=sharing

Is it the expected behaviour @mfrancisc?